src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheckTest.java

```diff
@@ -91,6 +91,31 @@ public void testCheckWithNoMatchingMethodName()
         verify(checkConfig, getPath("InputMatchXpath.java"), expected);
     }
 
+    @Test
+    public void testCheckWithSingleLineComments() throws Exception {
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(MatchXpathCheck.class);
+        checkConfig.addAttribute("query", "//SINGLE_LINE_COMMENT");
```

Comment:
ability to violate a token do not have real life value, please update query to use text of comment.

can we cover cases like:
https://github.com/checkstyle/checkstyle/blob/e3a9fa96c6294e50e21d2afdc604c243cc1385c4/config/checkstyle_checks.xml#L681-L685

Reply:
Done.

Reply:
In separate PR please substitute this hack by ToDoComment check, with better version by Xpath.

Reply:
Done, please see #10397.